hr_payslip_ytd_amount/report/report_payslip_ytd_amount.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding:utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Savoir-faire Linux. All Rights Reserved.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.report import report_sxw
+
+
+class payslip_ytd_amount_report(report_sxw.rml_parse):
+
+    def __init__(self, cr, uid, name, context):
+        super(payslip_ytd_amount_report, self).__init__(cr, uid, name, context)
+        self.localcontext.update({
+            'get_payslip_lines': self.get_payslip_lines,
+            'translate_term': self.translate_in_employee_lang,
+        })
+
+        self.employee_lang_dict = {}
+
+        payslips = self.pool['hr.payslip'].browse(
+            cr, uid, context.get('active_ids'), context=context)
+
+        for payslip in payslips:
+            employee = payslip.employee_id
+
+            user = employee.user_id
+
+            if employee.id not in self.employee_lang_dict:
+                if not user:
+                    # If the employee has no user, take the user that
+                    # requested the report
+                    user = self.employee_lang_dict[employee.id] = \
+                        self.pool['res.users'].browse(
+                            cr, uid, uid, context=context)
+
+                self.employee_lang_dict[employee.id] = user.lang
+
+    def get_payslip_lines(self, objects, context=None):
+        payslip_lines = self.pool.get('hr.payslip.line').browse(
+            self.cr, self.uid, [obj.id for obj in objects], context=context)
+
+        return [
+            line for line in payslip_lines
+            if line.appears_on_payslip and line.amount
+        ]
+
+    def translate_in_employee_lang(self, term, payslip, is_payslip_line=False):
```

Comment:
@dufresnedavid Thanks for the contribution.

What's wrong with the standard report translation function?